src/de/animestream/src/eu/kanade/tachiyomi/animeextension/de/animestream/AnimeStream.kt

```
@@ -64,77 +53,46 @@ class AnimeStream : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
-                .substringAfter("src=\"").substringBefore("\"")
-            when {
-                url.contains("https://streamtape") || url.contains("https://adblockeronstape") && hosterSelection?.contains("stape") == true -> {
-                    val quality = "Streamtape"
-                    val video = StreamTapeExtractor(client).videoFromUrl(url, quality)
-                    if (video != null) {
-                        videoList.add(video)
-                    }
-                }
-
-                url.contains("https://voe.sx") || url.contains("https://20demidistance9elongations.com") || url.contains("https://telyn610zoanthropy.com") && hosterSelection?.contains("voe") == true -> {
-                    val quality = "Voe"
-                    val video = VoeExtractor(client).videoFromUrl(url, quality)
-                    if (video != null) {
-                        videoList.add(video)
-                    }
-                }
-            }
+        val url = document.select("div a.lnk-lnk").attr("href")
+        val quality = "Metastream"
+        val video = MetaExtractor(client).videoFromUrl(url, quality)
+        if (video != null) {
+            videoList.add(video)
         }
         return videoList.reversed()
     }
 
-    override fun List<Video>.sort(): List<Video> {
-        val hoster = preferences.getString("preferred_hoster", null)
-        if (hoster != null) {
-            val newList = mutableListOf<Video>()
-            var preferred = 0
-            for (video in this) {
-                if (video.quality.contains(hoster)) {
-                    newList.add(preferred, video)
-                    preferred++
-                } else {
-                    newList.add(video)
-                }
-            }
-            return newList
-        }
-        return this
-    }
+    override fun List<Video>.sort(): List<Video> = throw Exception("not used")
```

Not sure if you can throw a Exception, `AnimeHttpSource` already has this method implemented, you can remove it

---

Correct, the extension will crash in this way.